modules/proc_deploy-quay-openshift-operator.adoc

```diff
@@ -1120,32 +1300,215 @@ run the operator locally:
+database migrations needed before scaling up the entire cluster.
+. Uninstall the old Quay Operator (v1.0.2 or older)
+. Install the latest Quay Operator (v3.3.0)
+. Create your CR by issuing the command `oc create -f new_deployment.yaml`
+. Watch the logs of your Quay Pod until all migrations have finished.
+. At this point, it is safe to scale up your Quay cluster if desired.
+
+=== Document the existing {productname} deployment
+
+For the purpose of ensuring a smooth upgrade, it is important to ensure you
+have all available configuration details *before* deleting your existing
+deployment. In the case that you must work with Red Hat Support, this
+information will aid them with the details needed to bring your cluster back
+to its original state. At minimum, the following information should be
+gathered:
+
+. The Custom Resource used to create your current Quay deployment.
+. The output of running `oc get QuayEcosystem -o yaml > quayecosystem.yaml`
+in your Project or Namespace.
+. The hostnames currently used to access Quay, Clair, Quay's Config App,
+Postgres, Redis, and Clair's Postgres instance. This can be achieved by
+executing: `oc get routes -o yaml > old_routes.yaml`
+. Any authentication details required to connect to your Postgres instance(s)
+for Quay and Clair.
+. Any authentication details required to connect to your data persistance
+provider such as AWS S3.
+. Backup your Quay's configuration secret which contains the `config.yaml`
+along with any certificates needed. This can be accomplished by using the
+following command:
++
+```
+$ oc get secret quay-enterprise-config-secret -o yaml > config-secret.yaml
+```
+
+=== Update the CR
 
-Before upgrading to a new version of {productname} or Clair, refer to the
-link:https://access.redhat.com/documentation/en-us/red_hat_quay/3/html-single/upgrade_red_hat_quay/index[Upgrade {productname}] guide for details.
-The instructions here tell you how to change the quay and clair-jwt containers,
-but do not provide the full upgrade instructions.
+Ensure a backup is created of your original Custom Resource (CR) before making any
+changes.
 
-At the point in the upgrade instructions where you are ready to identify the new quay and clair-jwt containers, here is what you do:
+If your deployment does not specify any specific network-related configuration
+values, this step may not be necessary. Please refer to the documentation to
+ensure that your the configuration options in your current CR are still
+accurate for the Quay Operator v3.3.0.
+
+In the case that you have specified options related to the management of
+networking, such as using a LoadBalancer or specifying a custom hostname,
+please reference the latest documentation to update them with the schema
+changes included in Quay Operator v3.3.0.
+
+If you have overridden the image used for Quay or Clair, please keep in mind
+that Quay Operator v3.3.0 specifically supports Quay v3.3.0 and Clair v????.
```

Comment:
Clair version?

Reply:
Ha! I meant to go back and fix that. It seems that Clair v4 is actually v3.3.0. I still need to figure out how to explain that one. I fixed it for now.